batch/batch/driver/job.py

```diff
@@ -200,6 +277,7 @@ async def mark_job_complete(
         return
 
     await notify_batch_job_complete(db, client_session, batch_id)
+    await notify_job_group_on_job_complete(db, client_session, batch_id, job_group_id)
```

Comment:
If this functionality is too much to add in this PR, I can easily rip this out and not allow a callback for child job groups. It can then be added in a separate PR.